<!-- jplaced
 2022, Universidad de Zaragoza -->

<!-- Launch file -->
<launch>
    <!-- New args -->
    <arg name="local_eta"   value="0.75"/>
    <arg name="global_eta"  value="1.0"/>
    <arg name="rate_detectors_hz"   value="10"/>
    <arg name="rate_filter_hz"      value="10"/>
    <arg name="rate_octo_hz"        value="1"/>
    <arg name="rate_grid_hz"        value="2"/>
    <arg name="map_topic_name" value="/map_iou_calculator/iou_map"/>
    <arg name="map_frame_id" value="iou_map"/>
    <arg name="gpu"  value="True"/>

    <arg name="map_topic_markers" value="/opencv_shapes_iou"/> 



    <arg name="color00"  value="1.0"/>
    <arg name="color01"  value="0.0"/>
    <arg name="color02"  value="0.0"/>


    <arg name="color10"  value="0.0"/>
    <arg name="color11"  value="1.0"/>
    <arg name="color12"  value="0.0"/>  


    <arg name="color20"  value="0.0"/>
    <arg name="color21"  value="0.0"/>
    <arg name="color22"  value="1.0"/>    


    
    <arg name="robot0_ns" value="robot_0"/>
    <arg name="agent_0" value="0"/>

    <arg name="robot1_ns" value="robot_1"/>
    <arg name="agent_1" value="1"/>    

    <arg name="robot2_ns" value="robot_2"/>
    <arg name="agent_2" value="2"/>



    <arg name="robot0_topic" value="/robot_0/filter/centroids"/>
    <arg name="robot1_topic" value="/robot_1/filter/centroids"/>
    <arg name="robot2_topic" value="/robot_2/filter/centroids"/>



    <arg name="map_topic1" default="/robot_0/gridmapper/projected_map"/>
    <arg name="map_topic2" default="/robot_1/gridmapper/projected_map"/> 
    <arg name="map_topic3" default="/robot_2/gridmapper/projected_map"/>



    <node name="map_iou_calculator" pkg="frontier_detector" type="frontier_detector_iou" output="screen">
        <param name="map_topic1" value="$(arg map_topic1)"/>
        <param name="map_topic2" value="$(arg map_topic2)"/>
    </node>

    <node pkg="frontier_detector" type="OpenCV_detector_iou.py" name="opencv_detector_iou" output="screen" respawn="true">
        <param name="map_topic" value="$(arg map_topic_name)"/>
        <param name="rate"      value="$(arg rate_detectors_hz)" />
        <param name="namespace"      value=""/>

    </node>

     <group ns="$(arg robot0_ns)">
        <node pkg="frontier_detector" type="Filter_iou.py" name="filter_iou" output="screen" respawn="true" respawn_delay="3">
        <param name="robot_topic" value="$(arg robot0_topic)"/>
        <param name="map_topic" value="$(arg map_topic_markers)"/>  
        <param name="distance_threshold" value="2"/> 

        <param  name="color0" value="$(arg color00)"/>
        <param  name="color1" value="$(arg color01)"/>
        <param  name="color2" value="$(arg color02)"/>
        <param name="color" value="[0.2,0.5,0]"/> 
        <param name="robot_ns" value="$(arg robot0_ns)"/>   
        </node>
    </group> 


    <group ns="$(arg robot1_ns)">
        <node pkg="frontier_detector" type="Filter_iou.py" name="filter_iou" output="screen" respawn="true" respawn_delay="3">
        <param name="robot_topic" value="$(arg robot1_topic)"/>
        <param name="map_topic" value="$(arg map_topic_markers)"/>     
         <param name="distance_threshold" value="2"/>
        <param  name="color0" value="$(arg color10)"/>
        <param  name="color1" value="$(arg color11)"/>
        <param  name="color2" value="$(arg color12)"/>
         <param name="robot_ns" value="$(arg robot1_ns)"/>        
        </node>
    </group> 




        
   <!-- 
        <node pkg="frontier_detector" type="frontier_detector_global_rrt_iou" name="global_detector" output="screen" respawn="true">
            <param name="map_topic" value="$(arg map_topic_name)"/>
            <param name="rate"      value="$(arg rate_detectors_hz)" />
            <param name="eta"       value="$(arg global_eta)"/>
        </node>


     -->




  
        <node pkg="frontier_detector" type="Initializer_iou.py" name="point_init" output="screen">
            <param name="rate" value="1.0"/>
            <param name="frame_id"      value="$(arg map_frame_id)"/>

        </node>


  
</launch>

